### ISSUE TITLE
Allow easier debugging of iOS SDK

### ISSUE THREAD

**Issue by petrdvorak** [lime-company/powerauth-mobile-sdk#76]
Debugging SDK internals in case something does not work is incredibly painful. We need to improve this. Good way could be to include source files with `PowerAuthSDK-Debug` CocoaPod, so that integrators can place a breakpoint in it and tell us more specifically where the issue is...

**Comment by hvge** [comment]
This is IMHO redundant to #49 From my own experience as an user of PA-SDK, the errors reported to the application is the biggest pain in the ass. If you need to debug the library, then the library is basically wrong by design. Which is not in my opinion, but we need to improve SDK behaviour when something's going off the rails.

**Comment by hvge** [comment]
Anyway, if we will do this, then we have to move C++ & "core" parts of SDK to the separate project file (e.g. separate static library), otherwise it will be hell on earth to force cocoapods to generate right projects with all dependencies on C++, cc7, openssl, etc... 

 It's more work that you would expect :)

**Comment by hvge** [comment]
The enhancement is not doable, unless the cocoapods fix its crappy dependency management. So, I'm adding a "sleeping" tag and will take a look on this when cocoapods 1.5 will be shipped.

### tl;dr

The blocker is our own support for IOS Extensions (currently in development) which is in conflict with regular PowerAuth library for IOS. We're sharing headers and some source codes in libraries for IOS, watchOS and IOS extensions and that's exactly what pod tool is not capable to handle.

### Long explanation

The pod tool re-generates project file for each `pod update` run. The main problem is that generated `Pods.xcodeproj` doesn't respect dependencies between pods and sets public pod's headers visible for each other pod used in the application (or application's target). This behaviour has a significant impact when both `PowerAuth2` and `PowerAuth2ForExtensions` pods are used in the application. These two libraries contains several shared source codes and importantly, are exposing several public headers with the same name. Due to fact that each pod can access headers from each other, the compiler then fails on duplicate interface definitions.

This happens independently on what pods are used in application's targets and leads actually to two different problems:

1. the problem with compiling pods as I described above
2. the problem with final application linking, when pod tool currently sets dependency on our both libraries to all available targets. That's really stupid and will be probably fixed in cocoapods 1.5

So, looks like that it will be easier to workaround current problems and later rewrite the high level codes to swift, which has a better module management. So that's why this is sleeping now...

**Comment by petrdvorak** [comment]
OK, makes sense.

**Comment by hvge** [comment]
Fixed by PR #96 

Now most of our SDK is distributed in pod as source code. Only low level C++ codes are precompiled into static FAT library, but those rarely contains bugs (due to good test coverage).
